node/core/provisioner/src/lib.rs

```diff
@@ -413,15 +428,24 @@ async fn select_candidates(
 
 		// we arbitrarily pick the first of the backed candidates which match the appropriate selection criteria
 		if let Some(candidate) = candidates.iter().find(|backed_candidate| {
-			let descriptor = &backed_candidate.candidate.descriptor;
+			let descriptor = &backed_candidate.descriptor;
 			descriptor.para_id == scheduled_core.para_id
 				&& descriptor.persisted_validation_data_hash == computed_validation_data_hash
 		}) {
-			selected_candidates.push(candidate.clone());
+			selected_candidates.push(candidate.hash());
 		}
 	}
 
-	Ok(selected_candidates)
+	// now get the backed candidates corresponding to these candidate receipts
+	let (tx, rx) = oneshot::channel();
+	sender.send(AllMessages::CandidateBacking(CandidateBackingMessage::GetBackedCandidates(
+		relay_parent,
+		selected_candidates,
+		tx,
+	)).into()).await.map_err(|err| Error::GetBackedCandidatesSend(err))?;
+	let candidates = rx.await.map_err(|err| Error::CanceledBackedCandidates(err))?;
```

Comment:
This should do a sanity check that the candidates are sorted in the correct order

Reply:
(ascending by assigned core index)

Either that or we should document that `GetBackedCandidates` may omit some of the candidates requested, but of those that it does return, they appear in the vec in the same order as requested.

Reply:
f0aee69